docs/proposals/config_data.md

```diff
@@ -175,16 +184,21 @@ type VolumeSource struct {
   ConfigData *ConfigDataVolumeSource `json:"configData,omitempty"`
 }
 
+// ConfigDataVolumeSource represents a volume that holds configuration data
 type ConfigDataVolumeSource struct {
-  Items []ConfigDataVolumeFile `json:"items,omitempty"`
+  // A list of config data keys to project into the volume in files
+  Files []ConfigDataVolumeFile `json:"files"`
 }
 
+// ConfigDataVolumeFile represents a single file containing config data
 type ConfigDataVolumeFile struct {
-  Path       string             `json:"path"`
-  ConfigData ConfigDataSelector `json:"configData"`
-}
-
+  ConfigDataSelector
```

Comment:
Needs `json:",inline"`

For example:
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L1998